apps/fhir/bluebutton/views/read.py

```diff
@@ -251,23 +240,14 @@ def generic_read(request,
 
     if interaction_type == 'search':
         if cx is not None:
-            # logger.debug("cx.fhir_id=%s" % cx.fhir_id)
             if cx.fhir_id.__contains__('/'):
                 id = get_fhir_id(cx).split('/')[1]
             else:
                 id = get_fhir_id(cx)
-            # logger.debug("Patient Id:%s" % r_id)
 
-    if resource_type.lower() == "patient":
-        key = get_fhir_id(cx)
-    else:
-        key = id
+    key = get_fhir_id(cx) if resource_type.lower() == "patient" else id
 
-    pass_params = build_params(pass_params,
-                               srtc,
-                               key,
-                               patient_id=get_fhir_id(cx)
-                               )
+    pass_params = urlencode(pass_params)
```

Comment:
The old version added a patient parameter, which the backend didn't like, and responded with a 400.

Given that the HAPI server fails on any unknown parameters, we need to switch to a whitelist approach, rather than passing any query parameter we see. Otherwise, extraneous query parameters will result in weird errors. Also increases are security testing surface area.

Also, the back end should probably be ignoring extraneous inputs rather than failing on them, but that's a different issue.

Reply:
For FHIR search operations, [the spec](https://www.hl7.org/fhir/search.html#errors) agrees with you, @samgensburg-gov:
> Servers may receive parameters from the client that they do not recognise, or may receive parameters they recognise but do not support (either in general, or for a specific search). In general, servers SHOULD ignore unknown or unsupported parameters ...

Further research needed to figure out why HAPI isn't honoring that and if it's reasonably fixable. I've filed [CBBF-105](https://issues.hhsdevcloud.us/browse/CBBF-105) on the backlog to address this at some point.

Reply:
Awesome, thanks for doing this Karl.